visma/functions/variable.py

```diff
@@ -83,7 +83,26 @@ def __radd__(self, other):
 
     def __add__(self, other):
         from visma.functions.constant import Constant
-        if isinstance(other, Constant):
+        if isinstance(other, Variable):
+            if self.power == other.power:
+                if self.before == '-':
+                    self.coefficient -= other.coefficient
+                else:
+                    self.coefficient += other.coefficient
+                result = Variable()
+                if self.coefficient != 0:
+                    result.scope = self.scope
+                    result.power = self.power
+                    result.coefficient = self.coefficient
+                return self
+            else:
```

Comment:
No need to put the tokens inside `Expression`. If addition is not possible, leave the expression as is.